src/main/java/com/google/sps/servlets/ImageHandlerServlet.java

```diff
@@ -0,0 +1,149 @@
+  private static final String ID_PARAMETER = "id";
+  private static final String IMAGE_PARAMETER = "image";
+  private static final String JPEG_PARAMETER = ".jpeg";
+  private static final String JPG_PARAMETER = ".jpg";
+  private static final String PLAYER_QUERY_PARAMETER = "Player";
+  private static final String PNG_PARAMETER = ".png";
+  private static final String UPLOADED_REDIRECT_PARAMETER = "/uploaded.html";
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Get the displayName entered by the user.
+    String displayName = request.getParameter(DISPLAY_NAME_PARAMETER);
+
+    // Get the URL of the image that the user uploaded to Blobstore.
+    String imageUrl = getUploadedFileUrl(request, IMAGE_PARAMETER);
+
+    Player currentPlayer = getCurrentPlayer();
+    if (currentPlayer != null) {
+      // Assign imageUrl to current player
+      currentPlayer.setImageID(imageUrl);
+
+      // Assign displayName to current player
+      currentPlayer.setDisplayName(displayName);
+    }
+
+    response.sendRedirect(UPLOADED_REDIRECT_PARAMETER);
+    ArrayList<Player> Players = PlayerDatabase.getPlayers();
+  }
+
+  // Gets the URL of the uploaded file; null if no file was uploaded
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+    // User submitted form without selecting a file
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+    // User submitted form without selecting a file (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      return null;
+    }
+    // Use ImagesService to get a URL that points to the uploaded file.
+    ImagesService imagesService = ImagesServiceFactory.getImagesService();
+    ServingUrlOptions options = ServingUrlOptions.Builder.withBlobKey(blobKey);
+    // To support running in Google Cloud Shell with AppEngine's dev server,
+    // we must use the relative
+    // path to the image, rather than the path returned by imagesService which contains a host.
+    try {
+      URL url = new URL(imagesService.getServingUrl(options));
+      return url.getPath();
+    } catch (MalformedURLException e) {
+      return imagesService.getServingUrl(options);
+    }
+  }
+
+  private boolean isValidImage(String imageUrl) {
+    if (imageUrl.endsWith(JPG_PARAMETER)
+        || imageUrl.endsWith(JPEG_PARAMETER)
+        || imageUrl.endsWith(PNG_PARAMETER)) {
+      return true;
+    }
+    return false;
```

Comment:
I think this can be broken down into a single line:
```
return imageUrl.endsWith(JPG_PARAMETER)
        || imageUrl.endsWith(JPEG_PARAMETER)
        || imageUrl.endsWith(PNG_PARAMETER);
```